cmd/lakectl/cmd/login.go

```diff
@@ -32,19 +38,99 @@ var (
 	loginRetryStatuses = slices.Concat(lakectlDefaultRetryStatuses,
 		[]int{http.StatusNotFound},
 	)
+
+	errNoProtocol = errors.New(`missing protocol, try e.g. "https://..."`)
+	errNoHost     = errors.New(`missing host, e.g. "https://honey-badger.lakefscloud.us-east-1.io/"`)
 )
 
 func loginRetryPolicy(ctx context.Context, resp *http.Response, err error) (bool, error) {
 	return CheckRetry(ctx, resp, err, loginRetryStatuses)
 }
 
+func validateURL(maybeURL string) error {
+	u, err := url.Parse(maybeURL)
+	if err != nil {
+		return err
+	}
+	switch {
+	case u.Scheme == "":
+		return errNoProtocol
+	case u.Host == "":
+		return errNoHost
+	default:
+		return nil
+	}
+}
+
+func isIPAddress(hostname string) bool {
+	return net.ParseIP(hostname) != nil
+}
+
+func readLoginServerURL() (string, error) {
+	var (
+		ok        = false
+		serverURL *url.URL
+		err       error
+	)
+	Write(warningNotificationTmpl, "No .lakectl.yaml found.  On lakeFS Enterprise, enter the server URL to log in.\n")
+	for !ok {
+		prompt := promptui.Prompt{
+			Label:    "lakeFS server URL",
+			Validate: validateURL,
+		}
+		serverURLString, err := prompt.Run()
+		if err != nil {
+			return "", err
+		}
+		serverURL, err = url.Parse(serverURLString)
+		if err != nil { // Unlikely, validateURL should have done this!
+			return "", err
+		}
+
+		host := serverURL.Hostname()
+		if isIPAddress(host) {
+			prompt = promptui.Prompt{
+				IsConfirm: true,
+				Default:   "n",
+				Label:     "Numeric IP addresses will not work with OIDC; are you sure? ",
+			}
+			_, err := prompt.Run()
+			if err != nil && !errors.Is(err, promptui.ErrAbort) {
+				return "", err
+			}
+			ok = !errors.Is(err, promptui.ErrAbort)
+		} else {
+			ok = true
+		}
+	}
+	return serverURL.String(), err
+}
+
+func configureLogin(serverURL string) error {
+	viper.Set("server.endpoint_url", serverURL)
+	return viper.SafeWriteConfig()
+}
+
 var loginCmd = &cobra.Command{
 	Use:     "login",
 	Short:   "Use a web browser to log into lakeFS",
 	Long:    "Connect to lakeFS using a web browser.",
 	Example: "lakectl login",
 	Run: func(cmd *cobra.Command, _ []string) {
-		serverURL, err := url.Parse(cfg.Server.EndpointURL.String())
+		var err error
+
+		writeConfigFile := false
+		serverEndpointURL := string(cfg.Server.EndpointURL)
+		if serverEndpointURL == "" {
+			serverEndpointURL, err = readLoginServerURL()
+			if err != nil {
+				DieFmt("No server endpoint URL: %s", err)
+			}
+			Write("Read URL {{. | yellow}}\n", serverEndpointURL)
```

Comment:
Text unclear, what does it mean "Read URL"? Maybe be more explicitly what your doing in here.